src/backends/mod.rs

```diff
@@ -199,6 +201,44 @@ pub trait Backend: Clone {
 
                 Ok(res)
             }
+            Value::HintIR(t, named_vars) => {
+                let mut precomp = PreComp::new();
+                // For hint evaluation purpose, precomp only has only one output and no connections with other parts,
+                // so just use a dummy output var name.
+                precomp.add_output("x".to_string(), t.clone());
+
+                // todo: encapsulate this in a field mapper
+                let cfg = CircCfg::default();
+                let cfg_f = cfg.field();
+
+                // map the named vars to env
+                let env = named_vars
+                    .iter()
+                    .map(|(name, var)| {
+                        let val = match var {
+                            // todo: convert ark_ff to rug::integer instead of u64
+                            crate::var::ConstOrCell::Const(cst) => cfg_f.new_v(cst.to_u64()),
```

Comment:
Is it safe to convert to u64 here?

Reply:
This should be field based. This will be fixed.